### PR TITLE
stackdriver audit log device

### PR DESCRIPTION
**Pull request by tam7t**
Adds support for a separate audit log device for writing logs directly to stackdriver. Similar to Issue #1027 but for Google Stackdriver instead of CloudWatch. 

Existing solutions (like https://github.com/sethvargo/vault-on-gke) require the user to write logs to stdout and rely on kubernetes to direct them to stackdriver or (https://github.com/terraform-google-modules/terraform-google-vault ) rely on writing them to disk and configure fluentd to transport them to Stackdriver.

Direct Stackdriver integration has the benefits:
- allow log writes synchronous to the request (instead of buffering to stdout or files)
- no additional sidecar processes to transport logs
- no need for filesystem rotation/cleanup

### CODE REVIEW

**Comment by jefferai** [comment]
Just a note on this: we're currently in the process of redesigning the auditing system so we're unlikely to merge this before that work is complete.

**Comment by tam7t** [comment]
Ah, sounds great. I noticed that there was quite a bit of duplication between the drivers around the hmac/logRaw/and salting. If there's demand for multiple providers I could see the plugin system being extended to support audit in addition to storage/secrets.

I'll leave this open and can update after the redesign.

**Comment by malnick** [comment]
After talking with @narayan-iyengar we decided to include this in our 1.5 release, still pending a formal review by the ecosystem team.

File: builtin/audit/stackdriver/backend.go [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+		Payload: json.RawMessage(buf.String()),
+		Resource: &mrpb.MonitoredResource{
+			Type: "generic_task",
+		},
+	}
+	if b.async {
+		b.lg.Log(log)
+		return nil
+	}
+	return b.lg.LogSync(ctx, log)
```
Comment (calvn) [inline builtin/audit/stackdriver/backend.go:184]: According to the [godocs](https://pkg.go.dev/cloud.google.com/go/logging?tab=doc#Logger.LogSync) `LogSync` blocks and can be slow and thus [not recommended](https://pkg.go.dev/cloud.google.com/go/logging?tab=doc#hdr-Synchronous_Logging) for normal usage, should this be a concern? I am not sure exactly how slow this translates to under normal operation (e.g. milliseconds vs seconds).

**Comment by tam7t** [comment]
Apologies, I forgot about this PR and no longer need the direct integration. Since #1027 is closed as well I will just close out this PR.

**Pull request closed without merging**